File: solutions_5648941810974720_0/Python/burakbugrul/kod.py
#-*- coding: UTF-8 -*-
import sys
from collections import Counter

temp = ("NINE")

case = int(raw_input())

for CASE in xrange(1, case+1):
	
	print "Case #{}:".format(CASE),

	inp = raw_input()
	cnt = Counter(inp)

	zero = cnt['Z']

	cnt['Z'] -= zero
	cnt['E'] -= zero
	cnt['R'] -= zero
	cnt['O'] -= zero

	six = cnt['X']

	cnt['S'] -= six
	cnt['I'] -= six
	cnt['X'] -= six

	eight = cnt['G']

	cnt['E'] -= eight
	cnt['I'] -= eight
	cnt['G'] -= eight
	cnt['H'] -= eight
	cnt['T'] -= eight

	three = cnt['H']

	cnt['T'] -= three
	cnt['H'] -= three
	cnt['R'] -= three
	cnt['E'] -= three * 2

	two = cnt['W']

	cnt['T'] -= two
	cnt['W'] -= two
	cnt['O'] -= two

	four = cnt['R']

	cnt['F'] -= four
	cnt['O'] -= four
	cnt['U'] -= four
	cnt['R'] -= four

	five = cnt['F']

	cnt['F'] -= five
	cnt['I'] -= five
	cnt['V'] -= five
	cnt['E'] -= five

	seven = cnt['V']

	cnt['S'] -= seven
	cnt['E'] -= seven * 2
	cnt['V'] -= seven
	cnt['N'] -= seven

	one = cnt['O']

	cnt['O'] -= one
	cnt['N'] -= one
	cnt['E'] -= one

	nine = cnt['E']

	ans = '0' * zero + '1' * one + '2' * two + '3' * three + '4' * four + '5' * five + '6' * six + '7' * seven + '8' * eight + '9' * nine
	print ans

File: solutions_5648941810974720_1/C++/aitch/kod.cpp
#include <iostream>
#include <algorithm>
#include <map>
#include <string>
#include <set>
#include <vector>
#include <queue>
#include <stack>
#include <cmath>
#include <bitset>
#include <random>
#include <cassert>
#include <cstdio>
#include <cstring>
using namespace std;
#define rep(i,a,b) for(int i = (a); i < int(b); ++i)
#define rrep(i,a,b) for(int i = (b); i --> int(a);)
#define allof(v) v.begin(),v.end()
#define pb push_back
typedef long long ll;
typedef vector<int> vi;
typedef vector<ll> vl;
typedef pair<int,int> pii;
typedef vector<bool> vb;
typedef long double ld;

int main(){
	srand(5);
	ios_base::sync_with_stdio(false);
	cin.tie(0);
	int T;
	cin >> T;
	rep(t,1,T+1){
		cout << "Case #" << t << ": ";
		string s;
		cin >> s;
		int cnt[26] = {};
		rep(i,0,26) cnt[i] = count(allof(s),'A'+i);
		vi ans(10);
		int perm[] = {0,6,8,3,2,4,7,5,1,9};
		string cc = "ZXGHWUSVOE";
		string namn[] = {"ZERO","ONE","TWO","THREE","FOUR","FVE","SIX","SEVEN","EIGHT","NINE"};
		rep(i,0,10){
			int d = perm[i];
			char c = cc[i];
			ans[d] = cnt[c-'A'];
			for(char cc : namn[d])
				cnt[cc-'A'] -= ans[d];
		}
		rep(i,0,10){
			rep(_,0,ans[i]) cout << char('0'+i);
		}
		cout << endl;
	}
}